test_main.py

```diff
@@ -66,30 +66,159 @@ def test_url_caching(empty_deployment):
+                    {
+                        "name": "x86_64_4_IOMMU_boot_test_dmesg.log",
+                        "url":
+                            "https://github.com/kernelci/kcidb/blob"
+                            "/main/requirements.txt?padding=1649"
+                    },
+                ],
+                "environment": {
+                    "comment": "meson-s905d-p230 in lab-baylibre",
+                    "misc": {
+                        "rootfs_url":
+                            "http://storage.kernelci/buildroot-baseline"
+                            "/20230120.0/armel/rootfs.cpio.gz"
+                    }
+                },
+            },
+        ],
+    }
+
+    # Submit data to submission queue
+    client.submit(data)
+
+    current_time = time.time()
+    # The time when caching system should be done with all our data
+    deadline_time = current_time + 120
+
+    # URLs to be check, as they should be cached
+    urls_expected = [
+        "https://github.com/kernelci/kcidb/blob/main/setup.py?padding=4673",
+        "https://github.com/kernelci/kcidb/blob/main/requirements.txt?"
+        "padding=1649",
+        "https://github.com/kernelci/kcidb/blob/main/.gitignore?"
+        "padding=547",
+        "https://github.com/kernelci/kcidb/blob/main/.pylintrc?padding=247",
+        "https://github.com/kernelci/kcidb/blob/main/doc/_index.md?"
+        "padding=7230",
+        "https://github.com/kernelci/kcidb/blob/main/.github/workflows/"
+        "deploy.yml?padding=4821",
+        "https://github.com/kernelci/kcidb/blob/main/doc/installation.md?"
+        "padding=1505"
     ]
 
-    for urls in urls_messages:
-        pub_client.publish(urls)
-
     # Retry checking URLs in the cache for a minute
     retry_interval = 5  # seconds
-    max_retries = 12  # 60 seconds / 5 seconds
-
-    for urls in urls_messages:
-        for url in urls:
-            for _ in range(max_retries):
-                if check_url_in_cache(url):
-                    break
-                time.sleep(retry_interval)
-            else:
+
+    for url in urls_expected:
+        while True:
+            if check_url_in_cache(url):
+                break
+            if time.time() > deadline_time:
                 raise AssertionError(f"URL '{url}' not found in the cache")
+            time.sleep(retry_interval)
+
+    current_time = time.time()
+    if current_time < deadline_time:
+        time.sleep(deadline_time - current_time)
+
+    # URL cases not to be cached
+    urls_not_expected = [
+        # Invalid url
+        'https://kernelcdn.kernel.org/pub/linux/',
+        # Larger-than-maximum size URL
+        "https://cdn.kernel.org/pub/linux/kernel/v6.x/"
+        "linux-6.4.11.tar.xz",
+        # Wrong hash URL
+        "https://github.com/kernelci/kcidb/blob/main/Dockerfile?"
+        "padding=200",
+        # URL from a wrong field
+        "http://storage.kernelci/buildroot-baseline/20230120.0/"
+        "armel/rootfs.cpio.gz",
```

Comment:
Please use a URL here, that is valid, works, and will stay available. Otherwise we won't be able to reliably say that it wasn't cached because of the wrong field (and not due to other reason).

Reply:
okay